langserver/symbol.go

```diff
@@ -68,14 +72,40 @@ func queryJoin(s, e string) string {
 	return strings.TrimSpace(s + " " + e)
 }
 
+func parseCaseSensitive(q string) (qu Query, fields []string) {
+	FieldName := "CaseSensitive"
+	FieldNamePrefix := fmt.Sprintf("%s:", FieldName)
+
+	for _, field := range strings.Fields(q) {
+		// Check if the field is a filter like `CaseSensitive:true`.
+		if !strings.HasPrefix(field, FieldNamePrefix) {
+			fields = append(fields, field)
+			continue
+		}
+
+		fieldVal := strings.TrimPrefix(field, FieldNamePrefix)
+		if caseSensitive, err := strconv.ParseBool(fieldVal); err != nil {
+			log.Printf("ParseQuery:CaseSensitive - parse error: %v", err)
+		} else {
+			qu.CaseSensitive = caseSensitive
+		}
+	}
+
+	return qu, fields
+}
+
 // ParseQuery parses a user's raw query string and returns a
 // structured representation of the query.
 func ParseQuery(q string) (qu Query) {
-	// All queries are case insensitive.
-	q = strings.ToLower(q)
+	// All queries are case insensitive, unless otherwise
+	qu, fields := parseCaseSensitive(q)
 
 	// Split the query into space-delimited fields.
-	for _, field := range strings.Fields(q) {
+	for _, field := range fields {
+		if !qu.CaseSensitive {
+			field = strings.ToLower(field)
+		}
+
```

Comment:
changed it around a bit and moved the conversion to a different func: `adjustCase`.
let me know if this clearer.